lib/pdk/util/bundler.rb

```diff
@@ -0,0 +1,129 @@
+require 'bundler'
+require 'tty-spinner'
+require 'pdk/util'
+require 'pdk/cli/exec'
+
+module PDK
+  module Util
+    module Bundler
+      class BundleHelper; end
+
+      def self.ensure_bundle!
+        bundle = BundleHelper.new
+
+        if bundle.has_gemfile?
+          if !bundle.locked?
+            bundle.lock
+          end
+
+          if !bundle.installed?
+            bundle.install
+          end
+        end
+      end
+
+      class BundleHelper
+        def has_gemfile?
+          # return a pure boolean
+          !!gemfile
+        end
+
+        def locked?
+          # return a pure boolean
+          !!gemfile_lock
+        end
+
+        def installed?
+          output_start(_("Checking for missing Gemfile dependencies"))
+
+          result = invoke('check', "--gemfile=#{gemfile}", "--path=#{pdk_cachedir}")
```

Comment:
I'd recommend changing the name of `pdk_cachedir` to something like `pdk_gemdir` to avoid confusion. In Bundler `cache` generally refers to the contents of `vendor/cache`, as opposed to the value of `--path` which is where the bundled gems are installed to.